core/tracepoint/path_spec.rb

```diff
@@ -9,12 +11,36 @@
       end
     end
 
-    it 'equals (eval) inside an eval' do
+    it 'equals (eval) inside an eval for :end event' do
       path = nil
       TracePoint.new(:end) { |tp| path = tp.path }.enable do
         eval("class A; end")
         path.should == '(eval)'
       end
     end
+
+    it 'equals the path of the file being run for :line event inside an eval' do
+      path = nil
+      TracePoint.new(:line) { |tp| path = tp.path }.enable do
+        eval("2 + 2")
+        path.should == "#{__FILE__}"
```

Comment:
I think that test is misleading. If I'm not mistaken, the block is probably called twice, the first time with `path` being `(eval)` and the second time being the current file. I think we can rely on the `:end` test and assume the other type of events will work the same.

Reply:
okay I have removed the specs for other events & updated the PR.